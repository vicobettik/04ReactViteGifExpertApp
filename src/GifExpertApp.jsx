import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';

export const GifExpertApp = () => {

  const [categories, setCategories] = useState(['one puch', 'dragon ball']);

  const onAddCategory = ( newCategory ) => {
    console.log(newCategory)
    // setCategories([newCategory, ...categories]);
    setCategories( (cat) =>  [newCategory, ...cat]);
  };

  return (
    <>
        {/* titulo */}
        <h1>Gif expert app</h1>

        {/* input  */}
        <AddCategory 
          // setCategories={ setCategories }
          onNewCategory={ (event) => onAddCategory(event) }
        />

        {/* listado de gifs */}
        <ol>
          {
            categories.map((cat) => {
              return <li key={cat}>{cat}</li>
            })
          }
        </ol>
          {/* gif item */}
    </>
  )
}

