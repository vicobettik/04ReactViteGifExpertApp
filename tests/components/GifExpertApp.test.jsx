import { fireEvent, render, screen } from "@testing-library/react"
import { GifExpertApp } from "../../src/GifExpertApp";

describe('Pruebas sobre GifExpertApp', () => {
  
    test('should ', () => {
      
        const { container } = render(<GifExpertApp />);
        expect(container).toMatchSnapshot();

        const input = screen.getByRole('textbox');
        fireEvent.input(input,
            {
                target:{
                    value: 'nezuko'
                }
            }
        );
        
        const form = screen.getByRole('form');
        fireEvent.submit(form);


        

        screen.debug();
        
    })
    

})
