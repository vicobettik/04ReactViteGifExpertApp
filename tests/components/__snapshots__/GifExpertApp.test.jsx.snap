// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre GifExpertApp should  1`] = `
<div>
  <h1>
    Gif expert app
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
</div>
`;
